server/src/jump_definition.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) ZeroC, Inc.
+
+use slicec::{
+    compilation_state::CompilationState,
+    grammar::{
+        Class, Commentable, CustomType, Entity, Enum, Enumerator, Exception, Field, Identifier,
+        Interface, MessageComponent, Module, NamedSymbol, Operation, Parameter, Struct, TypeAlias,
+        TypeRef, TypeRefDefinition, Types,
+    },
+    slice_file::{Location, SliceFile, Span},
+    visitor::Visitor,
+};
+use tower_lsp::lsp_types::{Position, Url};
+
+pub fn get_definition_span(state: &CompilationState, uri: Url, position: Position) -> Option<Span> {
+    // Attempt to convert the URL to a file path and then to a string
+    let file_path = uri.to_file_path().ok()?.to_str()?.to_owned();
+
+    // Attempt to retrieve the file from the state
+    let file = state.files.get(&file_path)?;
+
+    // Convert position to row and column to 1 based
+    let col = (position.character + 1) as usize;
+    let row = (position.line + 1) as usize;
+    let location: slicec::slice_file::Location = (row, col).into();
+
+    let mut visitor = JumpVisitor::new(location);
+    file.visit_with(&mut visitor);
+
+    visitor.found_span
+}
+
+struct JumpVisitor {
+    pub search_location: Location,
+    pub found_span: Option<Span>,
+}
+
+impl JumpVisitor {
+    pub fn new(search_location: Location) -> Self {
+        JumpVisitor {
+            search_location,
+            found_span: None,
+        }
+    }
+
+    fn check_comment(&mut self, commentable: &dyn Commentable) {
+        if let Some(comment) = commentable.comment() {
+            comment
+                .overview
+                .as_ref()
+                .map(|overview| self.check_message_links(&overview.message));
+            comment
+                .returns
+                .iter()
+                .for_each(|returns| self.check_message_links(&returns.message));
+            comment
+                .params
+                .iter()
+                .for_each(|params| self.check_message_links(&params.message));
+            comment
+                .throws
+                .iter()
+                .for_each(|throws| self.check_message_links(&throws.message));
+            comment
+                .throws
+                .iter()
+                .for_each(|s| self.check_and_set_span(s.thrown_type(), &s.span));
```

Comment:
Why not do these together in a single for loop, instead of iterating through them twice?
```suggestion
            for throws in &comment.throws {
                self.check_message_links(&throws.message)
                self.check_and_set_span(throws.thrown_type(), &s.span)
            }
```